.github/workflows/build-and-push-binaries.yml

```diff
@@ -1,8 +1,17 @@
 name: Build and Publish Binaries
 
 on:
+  workflow_call:
+    outputs:
+      matrix_info:
+        description: List of os / arch information in the build matrix
+        value: ${{ jobs.share_matrix_info.outputs.matrix_info }}
   workflow_dispatch:
     inputs:
+      publish_artifacts_to_release:
+        type: boolean
+        default: true
+        description: Upload binaries to an existing github release. When not set, binaries will be uploaded as temporary github artifacts.
```

Comment:
If you are interested, you could make the build_matrix an input to workflow_dispatch/workflow_call with the sensible default.